### ISSUE TITLE
Grakn Seems Down 

### ISSUE THREAD

**Issue by jemoussaoui** [OpenCTI-Platform/opencti#547]
Environment : 
   - Debian 10
   - OpenCTI 3.0.2

Problem : 
```
  error: [GRAKN] Seems down                                                                                                  
   error: [OPENCTI] GraphQL initialization fail > Error: Grakn seems down
```

**Comment by richard-julien** [comment]
Hi @jemoussaoui , can you upload here your docker compose, the result of docker ps command, and the logs of the grakn docker + opencti docker.
Thanks

**Comment by jemoussaoui** [comment]
docker-compose.yml

```
`version: '3'
services:
  grakn :
    image: graknlabs/grakn:1.6.2
    ports:
      - 48555:48555 
    volumes :
     - grakndata:/grakn-core-all-linux/server/db
  redis:
    image: redis:5.0.5
    restart: always
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.5.2
    volumes:
      - esdata:/usr/share/elasticsearch/data
    environment:
      - discovery.type=single-node
    restart: always
    ulimits:
      memlock:
        soft: -1
        hard: -1
      nofile:
        soft: 65536
        hard: 65536
  minio:
    image: minio/minio:RELEASE.2019-10-12T01-39-57Z
    volumes:
      - s3data:/data
    ports:
      - "9000:9000"
    environment:
      MINIO_ACCESS_KEY: ${MINIO_ACCESS_KEY}
      MINIO_SECRET_KEY: ${MINIO_SECRET_KEY}
    command: server /data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3
  rabbitmq:
    image: rabbitmq:3.7.17-management
    environment:
      - RABBITMQ_DEFAULT_USER=${RABBITMQ_DEFAULT_USER}
      - RABBITMQ_DEFAULT_PASS=${RABBITMQ_DEFAULT_PASS}    
    restart: always
  opencti:
    image: opencti/platform:latest
    environment:
      - APP__PORT=4000
      - APP__ADMIN__EMAIL=admin@opencti.io
      - APP__ADMIN__PASSWORD=....
      - APP__ADMIN__TOKEN=...
      - APP__LOGS_LEVEL=error
      - APP__LOGS=./logs
      - APP__REACTIVE=true
      - APP__COOKIE_SECURE=false
      - GRAKN__HOSTNAME=grakn
      - GRAKN__PORT=48555
      - GRAKN__TIMEOUT=30000
      - REDIS__HOSTNAME=redis
      - REDIS__PORT=6379
      - ELASTICSEARCH__URL=http://elasticsearch:9200
      - MINIO__ENDPOINT=minio
      - MINIO__PORT=9000
      - MINIO__USE_SSL=false
      - MINIO__ACCESS_KEY=...
      - MINIO__SECRET_KEY=...
      - RABBITMQ__HOSTNAME=rabbitmq
      - RABBITMQ__PORT=5672
      - RABBITMQ__PORT_MANAGEMENT=15672
      - RABBITMQ__MANAGEMENT_SSL=false
      - RABBITMQ__USERNAME=guest
      - RABBITMQ__PASSWORD=guest
      - PROVIDERS__LOCAL__STRATEGY=LocalStrategy
    ports:
      - "4000:4000"
    depends_on:
      - redis
      - elasticsearch
      - minio
      - rabbitmq
    restart: always
  worker:
    image: opencti/worker:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=...
      - WORKER_LOG_LEVEL=info
    depends_on:
      - opencti
    deploy:
      mode: replicated
      replicas: 3
    restart: always
  connector-export-file-stix:
    image: opencti/connector-export-file-stix:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=...
      - CONNECTOR_ID=... # Valid UUDv4
      - CONNECTOR_TYPE=INTERNAL_EXPORT_FILE
      - CONNECTOR_NAME=ExportFileStix2
      - CONNECTOR_SCOPE=application/json
      - CONNECTOR_CONFIDENCE_LEVEL=3
      - CONNECTOR_LOG_LEVEL=info
    restart: always
  connector-export-file-csv:
    image: opencti/connector-export-file-csv:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=..
      - CONNECTOR_ID=.. # Valid UUDv4
      - CONNECTOR_TYPE=INTERNAL_EXPORT_FILE
      - CONNECTOR_NAME=ExportFileCsv
      - CONNECTOR_SCOPE=application/csv
      - CONNECTOR_CONFIDENCE_LEVEL=3
      - CONNECTOR_LOG_LEVEL=info
    restart: always    
  connector-import-file-stix:
    image: opencti/connector-import-file-stix:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=...
      - CONNECTOR_ID=.. # Valid UUDv4
      - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
      - CONNECTOR_NAME=ImportFileStix2
      - CONNECTOR_SCOPE=application/json
      - CONNECTOR_CONFIDENCE_LEVEL=3
      - CONNECTOR_LOG_LEVEL=info
    restart: always
  connector-import-file-pdf-observables:
    image: opencti/connector-import-file-pdf-observables:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=..
      - CONNECTOR_ID=.. # Valid UUDv4
      - CONNECTOR_TYPE=INTERNAL_IMPORT_FILE
      - CONNECTOR_NAME=ImportFilePdfObservables
      - CONNECTOR_SCOPE=application/pdf
      - CONNECTOR_CONFIDENCE_LEVEL=3
      - CONNECTOR_LOG_LEVEL=info
      - PDF_OBSERVABLES_CREATE_INDICATOR=False
    restart: always    
  connector-opencti:
    image: opencti/connector-opencti:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=...
      - CONNECTOR_ID=.. # Valid UUDv4
      - CONNECTOR_TYPE=EXTERNAL_IMPORT
      - CONNECTOR_NAME=OpenCTI
      - CONNECTOR_SCOPE=identity,sector,region,country,city
      - CONNECTOR_CONFIDENCE_LEVEL=5
      - CONNECTOR_UPDATE_EXISTING_DATA=true
      - CONNECTOR_LOG_LEVEL=info
      - CONFIG_SECTORS_FILE_URL=https://raw.githubusercontent.com/OpenCTI-Platform/datasets/master/data/sectors.json
      - CONFIG_GEOGRAPHY_FILE_URL=https://raw.githubusercontent.com/OpenCTI-Platform/datasets/master/data/geography.json
      - CONFIG_INTERVAL=7 # Days
    restart: always
  connector-mitre:
    image: opencti/connector-mitre:latest
    environment:
      - OPENCTI_URL=http://localhost:4000
      - OPENCTI_TOKEN=..
      - CONNECTOR_ID=... # Valid UUDv4
      - CONNECTOR_TYPE=EXTERNAL_IMPORT
      - CONNECTOR_NAME=MITRE ATT&CK
      - CONNECTOR_SCOPE=identity,attack-pattern,course-of-action,intrusion-set,malware,tool,report
      - CONNECTOR_CONFIDENCE_LEVEL=3
      - CONNECTOR_UPDATE_EXISTING_DATA=true
      - CONNECTOR_LOG_LEVEL=info
      - MITRE_ENTERPRISE_FILE_URL=https://raw.githubusercontent.com/mitre/cti/master/enterprise-attack/enterprise-attack.json
      - MITRE_PRE_ATTACK_FILE_URL=https://raw.githubusercontent.com/mitre/cti/master/pre-attack/pre-attack.json
      - MITRE_INTERVAL=7 # Days
    restart: always
volumes:
  grakndata:
  esdata:
  s3data:`
```

docker ps : 

```
CONTAINER ID        IMAGE                                                  COMMAND                  CREATED             STATUS                            PORTS                                                 NAMES
5247ce8f1c57        opencti/connector-mitre:latest                         "/entrypoint.sh"         31 minutes ago      Up 2 seconds                                                                            docker_connector-mitre_1
397ac609e445        opencti/connector-export-file-stix:latest              "/entrypoint.sh"         31 minutes ago      Up 1 second                                                                             docker_connector-export-file-stix_1
9ab251929dd8        docker.elastic.co/elasticsearch/elasticsearch:7.5.2    "/usr/local/bin/dock…"   31 minutes ago      Up 4 seconds                      9200/tcp, 9300/tcp                                    docker_elasticsearch_1
a30910b1809d        opencti/connector-opencti:latest                       "/entrypoint.sh"         31 minutes ago      Up 2 seconds                                                                            docker_connector-opencti_1
d52eeb63c9f5        opencti/connector-import-file-pdf-observables:latest   "/entrypoint.sh"         31 minutes ago      Up 2 seconds                                                                            docker_connector-import-file-pdf-observables_1
d46008d9cc56        minio/minio:RELEASE.2019-10-12T01-39-57Z               "/usr/bin/docker-ent…"   31 minutes ago      Up 3 seconds (health: starting)   0.0.0.0:9000->9000/tcp                                docker_minio_1
19cc71a42d4f        opencti/connector-import-file-stix:latest              "/entrypoint.sh"         31 minutes ago      Up 2 seconds                                                                            docker_connector-import-file-stix_1
b755808efd03        opencti/connector-export-file-csv:latest               "/entrypoint.sh"         31 minutes ago      Up 3 seconds                                                                            docker_connector-export-file-csv_1
8cf74199c90c        graknlabs/grakn:1.6.2                                  "./grakn-docker.sh"      31 minutes ago      Up 4 seconds                      0.0.0.0:48555->48555/tcp                              docker_grakn_1
5f1332de7574        redis:5.0.5                                            "docker-entrypoint.s…"   31 minutes ago      Up 2 seconds                      6379/tcp                                              docker_redis_1
a3bb876a4d08        rabbitmq:3.7.17-management                             "docker-entrypoint.s…"   31 minutes ago      Up 2 seconds                      4369/tcp, 5671-5672/tcp, 15671-15672/tcp, 25672/tcp   docker_rabbitmq_1
```





docker logs docker_grakn_1 : 
```

====================================================================================================
      ________  _____     _______  __    __  __    __      _______  _______  _____     _______
     |   __   ||   _  \  |   _   ||  |  /  /|  \  |  |    |   _   ||   _   ||   _  \  |   ____|
     |  |  |__||  | |  | |  | |  ||  | /  / |   \ |  |    |  | |__||  | |  ||  | |  | |  |
     |  | ____ |  |_| /  |  |_|  ||  |/  /  |    \|  |    |  |     |  | |  ||  |_| /  |  |____
     |  ||_   ||   _  \  |   _   ||   _  \  |   _    |    |  |  __ |  | |  ||   _  \  |   ____|
     |  |__|  ||  | \  \ |  | |  ||  | \  \ |  | \   |    |  |_|  ||  |_|  ||  | \  \ |  |____
     |________||__|  \__\|__| |__||__|  \__\|__|  \__|    |_______||_______||__|  \__\|_______|

                                         THE KNOWLEDGE GRAPH
====================================================================================================
                                                                                      Version: 1.6.2
Starting Storage....FAILED!
Unable to start Storage.
====================================================================================================
      ________  _____     _______  __    __  __    __      _______  _______  _____     _______
     |   __   ||   _  \  |   _   ||  |  /  /|  \  |  |    |   _   ||   _   ||   _  \  |   ____|
     |  |  |__||  | |  | |  | |  ||  | /  / |   \ |  |    |  | |__||  | |  ||  | |  | |  |
     |  | ____ |  |_| /  |  |_|  ||  |/  /  |    \|  |    |  |     |  | |  ||  |_| /  |  |____
     |  ||_   ||   _  \  |   _   ||   _  \  |   _    |    |  |  __ |  | |  ||   _  \  |   ____|
     |  |__|  ||  | \  \ |  | |  ||  | \  \ |  | \   |    |  |_|  ||  |_|  ||  | \  \ |  |____
     |________||__|  \__\|__| |__||__|  \__\|__|  \__|    |_______||_______||__|  \__\|_______|

                                         THE KNOWLEDGE GRAPH
====================================================================================================
                                                                                      Version: 1.6.2

Starting Storage....FAILED!
Unable to start Storage.
```


docker logs docker_opencti_1 : 

```
yarn run v1.19.1
$ node --max_old_space_size=8192 build/index.js
error: [GRAKN] Seems down
error: [OPENCTI] GraphQL initialization fail > Error: Grakn seems down
    at B (/opt/opencti/build/index.js:1:5498)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
yarn run v1.19.1
```

**Comment by richard-julien** [comment]
Thanks.
Seems to be the same problem as https://github.com/OpenCTI-Platform/opencti/issues/542 with grakn that fail to start in docker environment. An issue is currently open in grakn side https://github.com/graknlabs/grakn/issues/5570.

The current workaround seems to start grakn alone before starting the rest of the dependencies. I know this is a bit annoying but can you try this workaround? (start grakn alone, check if start success and then start other containers).

Thanks

**Comment by darcosion** [comment]
> 
> 
> Thanks.
> Seems to be the same problem as #542 with grakn that fail to start in docker environment. An issue is currently open in grakn side [graknlabs/grakn#5570](https://github.com/graknlabs/grakn/issues/5570).
> 
> The current workaround seems to start grakn alone before starting the rest of the dependencies. I know this is a bit annoying but can you try this workaround? (start grakn alone, check if start success and then start other containers).
> 
> Thanks

I've tried this workaround because I've the same error (https://github.com/OpenCTI-Platform/opencti/issues/137#issuecomment-594008433) and it's work. So maybe putting a sleep after starting grakn for the rest of services ? 

Regards

**Comment by darcosion** [comment]
By the way, when it start, I have many `OpenCTI API seem down`, it's related ? Should I need to open a new issue for that ? 

Regards

**Comment by richard-julien** [comment]
Maybe a sleep can work but grakn have to fix this :) For the `OpenCTI API seem down` its a normal situation because the openCTI API cannot start if Grakn is not up. The strategy is to let the manager (docker compose, kubernetes, ...) handle the problematic of starting the containers. I understand it can be strange to have some failure until everything start to work but its the current strategy. So no need to create an issue for that :)

**Comment by darcosion** [comment]
But Grakn isn't down for me, I've fixed that with a separate start of Grakn, so the OpenCTI API should be up ? 

I've left the docker-compose start process a long time, but OpenCTI seem always down, and the web UI is accessible.

**Comment by richard-julien** [comment]
Oh ok. Can you check the docker logs for opencti ? Thanks

**Comment by darcosion** [comment]
I use this methodology to get logs, its a good way ? 

1. sudo docker container ls #get the container id of opencti/plateform:latest
2. sudo docker logs [container id]

And the result : 
```


    yarn run v1.19.1
    $ node --max_old_space_size=8192 build/index.js
    error: [ELASTICSEARCH] Seems down
    error: [OPENCTI] GraphQL initialization fail > Error: elastic seems down
        at D (/opt/opencti/build/index.js:1:41940)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    yarn run v1.19.1
    $ node --max_old_space_size=8192 build/index.js
    error: [ELASTICSEARCH] Seems down
    error: [OPENCTI] GraphQL initialization fail > Error: elastic seems down
        at D (/opt/opencti/build/index.js:1:41940)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    yarn run v1.19.1
    $ node --max_old_space_size=8192 build/index.js
    error: [ELASTICSEARCH] Seems down
    error: [OPENCTI] GraphQL initialization fail > Error: elastic seems down
        at D (/opt/opencti/build/index.js:1:41940)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
    yarn run v1.19.1
    $ node --max_old_space_size=8192 build/index.js
    error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
    error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
    error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
    error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
    error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
```

**Comment by richard-julien** [comment]
Yes thats fine to get the logs.
Looks like Elasticsearch have some problem to start `error: [ELASTICSEARCH] Seems down`
Can you also check the logs of elastic and try to see why it fail to start? Maybe not enough memory or other boring stuff :)

**Comment by darcosion** [comment]
Done, It seem Elasticsearch had many "All shared failed", so I've followed this stackoverflow solutions : https://stackoverflow.com/questions/21157466/all-shards-failed
```
curl -XPUT -H 'Content-Type: application/json' 'localhost:9200/_settings' -d '
{
    "index" : {
        "number_of_replicas" : 0
    }
}'
```
Now, Elasticsearch work and I've done a ``sudo docker restart [container_id]`.
In the log, I have just : 
```
yarn run v1.19.1
$ node --max_old_space_size=8192 build/index.js
```

But I have already same troubles with OpenCTI API.

**Comment by darcosion** [comment]
Having again logs like : 
```
error: [ELASTICSEARCH] Paginate fail > ResponseError: search_phase_execution_exception
```

But I think it's not relevant because ElasticSearch is now up.

**Comment by richard-julien** [comment]
This error is not a problem. We create some elastic structure at the first element creation, so you can have this kind of error until you create some entities in openCTI, I will modify the severity to warning for the next release

**Comment by richard-julien** [comment]
Can you confirm @darcosion the platform is now running? Thanks

**Comment by darcosion** [comment]
Unfortunately, I've again `OpenCTI API seem down`, I retry tomorrow but I think I should open a new ticket because Grakn work, it's more related to my other ticket about connexion to RabbitMQ impossible : https://github.com/OpenCTI-Platform/connectors/issues/54#issue-576241165

**Comment by jemoussaoui** [comment]
Hi @richard-julien 
i've tried the manual installation,
and i got this problem


```
yarn serv
yarn run v1.22.0
$ node --max_old_space_size=8192 build/index.js
info: 🚀 OpenCTI started in production mode with embedded file
info: [PRE-CHECK] > Grakn is alive
info: [PRE-CHECK] > Elasticsearch is alive
error: [OPENCTI] GraphQL initialization fail > S3Error: The access key ID you provided does not exist in our records.
    at Object.parseError (/root/app/opencti/node_modules/minio/dist/main/xml-parsers.js:85:11)
    at /root/app/opencti/node_modules/minio/dist/main/transformers.js:155:22
    at DestroyableTransform._flush (/root/app/opencti/node_modules/minio/dist/main/transformers.js:79:10)
    at DestroyableTransform.prefinish (/root/app/opencti/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:129:10)
    at DestroyableTransform.emit (events.js:311:20)
    at prefinish (/root/app/opencti/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:611:14)
    at finishMaybe (/root/app/opencti/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:620:5)
    at endWritable (/root/app/opencti/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:643:3)
    at DestroyableTransform.Writable.end (/root/app/opencti/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:571:22)
    at IncomingMessage.onend (_stream_readable.js:679:10)
    at Object.onceWrapper (events.js:417:28)
    at IncomingMessage.emit (events.js:323:22)
    at endReadableNT (_stream_readable.js:1204:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by richard-julien** [comment]
Hi @jemoussaoui, please come to the slack channel to ask your question or create a new ticket for this question. (In manual installation you have to configure a config file that must match that you configure in minio in term of credentials.)

**Comment by darcosion** [comment]
@richard-julien , I confirm you than my platform has again `OpenCTI API seem down` trouble. Well... Actually, I have no idea of resolutions or investigations. Should I tcpdump trafic on opencti-platform container ? Or maybe it's related to AQMP errors ? So going to RabbitMQ logs ?

**Comment by richard-julien** [comment]
Can you join the slack? Will be easier to continue the discussion

**Comment by darcosion** [comment]
Hello, 

Found a solution with Grakn : 
```
docker container ls #searching the id of Grakn container
docker restart [id_docker_grakn]
```
If you'r on Ubuntu, a sudo before docker could be specified. ;)

For verifying if Grakn work : 
```
docker logs [id_docker_grakn]
```

**Comment by mathurin68** [comment]
Hmmm, what are you supposed to do if you're just running 'docker-compose up' ... there's nothing running or created to restart?

**Comment by SamuelHassine** [comment]
Do you still have this bug with the latest version & Grakn 1.7.1?

**Comment by darcosion** [comment]
Hello, 

I need to reinstall the plateform, so i'm going to return on this issue after this installation saying if the bug is always here. ;)

**Comment by SamuelHassine** [comment]
Hello, I close this issue due to inactivity, if you still have a problem please re-open it!

**Comment by darcosion** [comment]
Hello, 

I've done my reinstall, it's work perfectly, sorry for the latency. ^^'

Regards,
